axolotls/list_column.py

```diff
@@ -35,13 +35,22 @@ def __getitem__(self, key):
             return None
 
         if isinstance(key, slice):
+            # non-continugous slice is not supported yet
+            assert key.step is None
+            start = key.start or 0
+            stop = key.stop or len(self)
+
+            values = self.values[self.offsets[start] : self.offsets[stop]]
+
             # convert offsets to length for selection
-            lengths = self.offsets[1:] - self.offsets[:-1]
+            lengths: torch.Tensor = self.offsets[1:] - self.offsets[:-1]
             lengths = lengths[key]
             # convert selected lengths back to offsets
-            offsets = torch.cat((torch.tensor([0]), torch.cumsum(lengths, dim=0)))
+            offsets = torch.cat((
+                torch.tensor([0], dtype=torch.int32),
+                torch.cumsum(lengths, dtype=torch.int32, dim=0)
+            ))
 
-            values = self.values[key]
```

Comment:
Fix the bug previously :) . Needs to use `self.values[self.offsets[start] : self.offsets[stop]]` (so it's to slice over the "flattened" array~)